docs/src/main/sphinx/security/apache-ranger-access-control.md

```diff
@@ -0,0 +1,168 @@
+
+  <property>
+    <name>ranger.plugin.trino.access.cluster.name</name>
+    <value></value>
+    <description>Name to identify the cluster running the Trino instance. This is recorded in audit logs generated by the plugin</description>
+  </property>
+
+  <property>
+    <name>ranger.plugin.trino.use.rangerGroups</name>
+    <value>false</value>
+    <description>Boolean flag to specify whether user-to-groups mapping should be obtained from in Apache Ranger. Default: false</description>
+  </property>
+
+  <property>
+    <name>ranger.plugin.trino.use.only.rangerGroups</name>
+    <value>false</value>
+    <description>Boolean flag. true: use user-to-groups mapping from Apache Ranger; false: use user-to-groups mappings from Apache Ranger and Trino. Default: false</description>
+  </property>
+
+  <property>
+    <name>ranger.plugin.trino.super.users</name>
+    <value></value>
+    <description>Comma separated list of user names. Superusers will be authorized for all accesses, without requiring explicit policy grants.</description>
+  </property>
+
+  <property>
+    <name>ranger.plugin.trino.super.groups</name>
+    <value></value>
+    <description>Comma separated list of group names. Users in supergroups will be authorized for all accesses, without requiring explicit policy grants</description>
+  </property>
+</configuration>
+```
+
+### ranger-trino-audit.xml
+```
+<?xml-stylesheet type="text/xsl" href="configuration.xsl"?>
+<configuration xmlns:xi="http://www.w3.org/2001/XInclude">
+  <property>
+    <name>xasecure.audit.is.enabled</name>
+    <value>true</value>
+    <description>Boolean flag to specify if the plugin should generate access audit logs. Default: true</description>
+  </property>
+
+  <property>
+    <name>xasecure.audit.solr.is.enabled</name>
+    <value>false</value>
+    <description>Boolean flag to specify if audit logs should be stored in Solr. Default: false</description>
+  </property>
+
+  <property>
+    <name>xasecure.audit.solr.solr_url</name>
+    <value></value>
+    <description>URL to Solr deployment where the plugin should send access audits to</description>
+  </property>
+</configuration>
+```
+
+### ranger-policymgr-ssl.xml
+```
+<?xml-stylesheet type="text/xsl" href="configuration.xsl"?>
+<configuration xmlns:xi="http://www.w3.org/2001/XInclude">
+  <!-- properties used for 2-way SSL between the Trino plugin and Apache Ranger server -->
+  <property>
+    <name>xasecure.policymgr.clientssl.keystore</name>
+    <value></value>
+    <description>Path to keystore file</description>
+  </property>
+
+  <property>
+    <name>xasecure.policymgr.clientssl.keystore.type</name>
+    <value>jks</value>
+    <description>Type of keystore. Default: jks</description>
+  </property>
+
+  <property>
+    <name>xasecure.policymgr.clientssl.keystore.credential.file</name>
+    <value></value>
+    <description>Path to credential file for the keystore; the credential should be in alias sslKeyStore</description>
```

Comment:
I'm struggling to understand "the credential should be in alias sslKeyStore"

Some documentation (or pointers to relevant documentation) on how to create and maintain these would be useful here. 

Additionally is this requiring the credential file (which holds the password to the keystore?) which is essentially the secret to be supplied on the server?

typically trino uses the ${ENV:SECRETNAME} notation to deal with secrets as detailed here https://trino.io/docs/current/security/secrets.html

Reply:
Additionally, i think the format here needs to be something like 
jceks://file/path/to/file/cred.jceks

without any example this is not clear

Reply:
@mneethiraj digging into this, the cred file seems to have previously been created as part of the plugin install in enable-trino-plugin.sh  useing org.apache.ranger.credentialapi.buildks from the ranger jars

Given that that no longer exists we're going to need details of how this can be created

Reply:
> the cred file seems to have previously been created as part of the plugin install in enable-trino-plugin.sh useing org.apache.ranger.credentialapi.buildks from the ranger jars
> 
> Given that that no longer exists we're going to need details of how this can be created

Java `keytool` can be used instead of `buildks` utility included in Ranger. [JDK documentation on keytool](https://docs.oracle.com/en/java/javase/11/tools/keytool.html) has necessary details on using it to manage keystores.

Reply:
Again, could/should this be updated updates to use the secrets from env, this is inconsistent with 
```
http-server.https.truststore.path=/mnt/shared/certs/localcacerts
http-server.https.truststore.key=<truststore-password>
```
Detailed here https://trino.io/docs/current/security/inspect-jks.html

@kokosing @mosabua what do you think